SLA liquidity (#1685)
vegaprotocol/specs: protocol/0035-LIQM-liquidity_monitoring.md

Note that as long as all pegs that LP batch orders can peg to exists on the book there is one-to-one correspondence between the total stake committed by liquidity providers (LPs), see [LP mechanics](./0044-LIME-lp_mechanics.md) spec, and the total supplied liquidity.
Indeed

`lp_liquidity_obligation_in_ccy_volume = market.liquidity.stakeToCcyVolume ⨉ stake`.

> Thus it is sufficient to compare `target_stake`.

Reviewer: ```suggestion
Thus it is sufficient to compare `target_stake` with `total_stake`. 
```

### Decreasing supplied stake

> If the [liquidity provision transaction would decrease](./0044-LIME-lp_mechanics.md#liquidity-provider-proposes-to-amend-commitment-amount) `supplied_stake` so that liquidity auction gets triggered then a liquidity auction is triggered the next time the next time conditions for liquidity auctions are evaluated.

Reviewer 2: ```suggestion
If the [liquidity provision transaction would decrease](./0044-LIME-lp_mechanics.md#liquidity-provider-proposes-to-amend-commitment-amount) `supplied_stake` so that liquidity auction gets triggered then a liquidity auction is triggered the next time conditions for liquidity auctions are evaluated.
```

> If the [liquidity provision transaction would decrease](./0044-LIME-lp_mechanics.md#liquidity-provider-proposes-to-amend-commitment-amount) `supplied_stake` so that liquidity auction gets triggered then a liquidity auction is triggered the next time the next time conditions for liquidity auctions are evaluated. 

Reviewer: No more lock-in at all then? I would worry the combination of no lock in and still having liquidity auctions might mean in a crashing market the LPs all pull out and leave people with positions in the market trapped in a liquidity auction?

Reviewer: Might it be possible for liquidity auctions to periodically allow uncrossing of some trades which reduce market OI without actually exiting the auction (I.e it reduces but just not enough to drop below target)

Reviewer: That might breach the simplicity rules though..

Reviewer 3: > No more lock-in at all then? I would worry the combination of no lock in and still having liquidity auctions might mean in a crashing market the LPs all pull out and leave people with positions in the market trapped in a liquidity auction?

No lock in but you do lose some or all of your bond for quitting if below target stake.

Reviewer 3: > Might it be possible for liquidity auctions to periodically allow uncrossing of some trades which reduce market OI without actually exiting the auction (I.e it reduces but just not enough to drop below target)

I like the idea of an uncrossing where the market remains in auction after uncrossing, that can only occur if OI is reduced. @TomMcL would you be able to take a look at how this would fit with the relevant liquidity monitoring and auction specs?

> If the `supplied_stake` decreases as a result of a closeout of an insolvent liquidity provider, then closeout should proceed and market should go into liquidity auction the next time the next time conditions for liquidity auctions are evaluated.

Reviewer 2: ```suggestion
If the `supplied_stake` decreases as a result of a closeout of an insolvent liquidity provider, then closeout should proceed and market should go into liquidity auction the next time conditions for liquidity auctions are evaluated.
```

## Trigger for exiting the auction

We exit if

`total_stake >= target_stake`.

> During the liquidity monitoring auction new or existing LPs can commit more stake (and hence liquidity) through the special market making order type and enable this by posting enough margin - see the [liquidity provision mechanics](./0044-LIME-lp_mechanics.md) spec for details. These need to be monitored to see if auction mode can be exitted.

Reviewer 2: ```suggestion
During the liquidity monitoring auction new or existing LPs can commit more stake (and hence liquidity) through the special market making transaction and enable this by posting enough margin - see the [liquidity provision mechanics](./0044-LIME-lp_mechanics.md) spec for details. These need to be monitored to see if auction mode can be exited.
```
Wondering if it still makes sense to refer to these as "order types", though it would probably require a broader renaming across other specs.

Reviewer 3: Agree this should be a transaction (I guess "liquidity commitment transaction") now.
